client/src/components/AddNewItem/AddNewItem.tsx

```diff
@@ -11,18 +28,90 @@ interface Props {
 export default function AddNewItem({ lists }: Props): JSX.Element {
   const classes = useStyles();
   const [list, setList] = useState('');
+  const [openNewItemModal, setOpenNewItemModal] = useState(false);
+  const [newItemUrl, setNewItemUrl] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [snackbar, setSnackbar] = useState({
+    openSnackbar: false,
+    notification: '',
+    severity: 'info',
+  });
+  const [product, setProduct] = useState({
+    name: '',
+    price: '',
+    imageUrl: '',
+  });
+
+  function Alert(props: AlertProps) {
+    return <MuiAlert elevation={6} variant="filled" {...props} />;
+  }
 
   const handleChange = (event: React.ChangeEvent<{ value: unknown }>) => {
     setList(event.target.value as string);
   };
 
+  const handleUrlChange = (event: React.ChangeEvent<{ value: unknown }>) => {
+    setNewItemUrl(event.target.value as string);
+  };
+
+  const handleOpen = () => {
+    if (!isLoading) {
+      if (newItemUrl == '') {
+        setSnackbar({
+          openSnackbar: true,
+          notification: 'Enter a product URL first!',
+          severity: 'info',
+        });
+      } else if (list == '') {
+        setSnackbar({
+          openSnackbar: true,
+          notification: 'Please select a list!',
+          severity: 'info',
+        });
+      } else {
+        setIsLoading(true);
+        createNewProduct(newItemUrl, list).then((res: ProductInterface) => {
+          setIsLoading(false);
+          setProduct({
+            name: res.name,
+            price: res.price,
+            imageUrl: res.imageUrl,
+          });
+          setOpenNewItemModal(true);
+        });
+      }
+    }
+  };
+
+  const handleClose = () => {
+    setOpenNewItemModal(false);
+    setSnackbar({
+      openSnackbar: true,
+      notification: 'Product added to list',
+      severity: 'success',
+    });
+  };
+
+  const handleSnackbarClose = () => {
+    setSnackbar({
+      openSnackbar: false,
+      notification: '',
+      severity: 'info',
+    });
+  };
+
   return (
     <Grid container justify="center" alignItems="center" direction="column" className={classes.root}>
       <Typography variant="h5">
         <Box fontWeight={700}>Add new item:</Box>
       </Typography>
       <form className={classes.form}>
-        <Input disableUnderline={true} placeholder="Paste your link here" className={classes.input}></Input>
+        <Input
+          disableUnderline={true}
+          placeholder="Paste your link here"
+          className={classes.input}
+          onChange={handleUrlChange}
+        ></Input>
```

Comment:
```suggestion
         / >
```

since there's no child component